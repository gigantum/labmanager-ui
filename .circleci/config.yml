version: 2
jobs:
  build:
    docker:
      - image: gigantum/circleci-service-client:992bcbab-2018-07-09
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: Initialize submodules
          command: |
            git submodule update --init --recursive
      - run:
          name: Build frontend production code
          command: |
            npm install
            npm install relay
            npm run relay
            unset CI && npm run build
      - run:
          name: Initialize submodules
          command: |
            jest dashboard -u
