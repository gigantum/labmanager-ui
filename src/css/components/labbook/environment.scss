.Environment{
  overflow: auto;
  height: calc(100% - 13px);
  width: 100%;
}

.Environment__header-container{
  @include flex(left, row)
}

.Environment__header{
  font-weight: light;
  color: $font-grey;
  font-size: 22px;
}

.Environment__edit-button,
.Environment__edit-button:hover,
.Environment__edit-button:focus{
  font-weight: light;
  color: $font-grey;
  font-size: 14px;
  height: 30px;
  width: 30px;
  padding: 5px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}

.Environment__card{
  background-color: $white;
  padding: 10px;
  color: $charcoal;
  width: 330px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
}


.Environment__image-container{
  align-items: center;
  align-content: center;
  align-self: center;
}

.Environment__card-text{
  //background-color: $grey-light;
  padding: 10px;
  width: 200px;
}

.Environment__edit-container{
  align-self: center;
  margin-left: 20px;
}

.Environment__modal{
  position: fixed;
  left: 50%;
  top: 50%;
  z-index: 100;
  width: 800px;
  height: 600px;

  margin: -300px 0 0 -400px;

  border: 1px solid $grey;
  background-color: $grey-light;
  box-shadow: 10px 10px 5px #888888;

  .SelectBaseImage__progress-button{
    display: none;
  }

}

.Environment__modal-close{
  position: absolute;
  top: 5px;
  right: 5px;
  cursor: pointer;
}

.Environment__package-dependencies{
  margin: 10px 20px 10px 0;

  .Environment__card-text{
      background-color: $white;
      width: 150px;
      box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
      p{
        margin: 0 5px 0 0;
      }

  }
}

.Environment__development-environment-item,
.Environment__dependencies{
  margin: 0 20px 0 0;
}
