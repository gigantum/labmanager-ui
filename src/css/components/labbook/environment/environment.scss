.Environment {
  width: 100%;
}

.Environment__header-container {
  @include flex(left, row)
}

.Environment__header {
  font-weight: light;
  color: $font-grey;
  font-size: 22px;
}

.Environment__edit-button,
.Environment__edit-button:hover,
.Environment__edit-button:focus {
  font-weight: light;
  color: $font-grey;
  font-size: 14px;
  height: 30px;
  width: 30px;
  padding: 5px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}

.Environment__card {
  background-color: $white;
  padding: 10px;
  color: $jet;
  width: 100%;
  @extend .box-shadow;
}


.Environment__image-container {
  align-items: center;
  align-content: center;
  align-self: center;
}

.Environment__card-text {
  //background-color: $grey-light;
  padding: 10px;
  width: 200px;
}

.Environment__edit-container {
  align-self: center;
  margin-left: 20px;
}

.Environment__modal {
  position: fixed;
  @include modal(500px, 800px);

  .SelectDevelopmentEnvironment__title,
  .AddEnvironmentPackage__title {
    font-size: 1.5em;
    text-align: center;
    border-bottom: 1px solid $azure;
    padding: 5px;
    color: $azure;
  }

  .SelectBaseImage__progress-button {
    display: none;
  }

}

.Environment__modal-close {
  cursor: pointer;
  @include close();
}

.Environment__package-dependencies {
  margin: 10px 20px 10px 0;

  .Environment__card-text{
      background-color: $white;
      width: 150px;
      @extend .box-shadow;

      p{
        margin: 0 5px 0 0;
      }

  }
}

.Environment__development-environment-item,
.Environment__dependencies {
  margin: 0 20px 0 0;
}
