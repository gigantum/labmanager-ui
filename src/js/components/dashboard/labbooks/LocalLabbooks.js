//vendor
import React, { Component } from 'react'
import {
  createPaginationContainer,
  graphql
} from 'react-relay'
//components
import WizardModal from 'Components/wizard/WizardModal'
import Loader from 'Components/shared/Loader'
import LocalLabbookPanel from 'Components/dashboard/labbooks/LocalLabbookPanel'
import ImportModule from 'Components/import/ImportModule'




let isLoadingMore = false;

class LocalLabbooks extends Component {

  constructor(props){
  	super(props);

    this._goToLabbook = this._goToLabbook.bind(this)
    this._loadMore = this._loadMore.bind(this)

  }

  /**
  * fires when a componet mounts
  * adds a scoll listener to trigger pagination
  */
  componentDidMount() {
    let that = this;
    window.addEventListener('scroll', function(e){
      let root = document.getElementById('root')
      let distanceY = window.innerHeight + document.documentElement.scrollTop + 200,
          expandOn = root.offsetHeight;

      if(that.props.feed.localLabbooks){
        if ((distanceY > expandOn) && !isLoadingMore && that.props.feed.localLabbooks.pageInfo.hasNextPage) {
            that._loadMore(e);
        }
      }
    });
  }
  /**
  *  @param {string} labbookName - inputs a labbook name
  *  routes to that labbook
  */
  _goToLabbook = (labbookName) => {
    this.setState({'labbookName': labbookName})

    this.props.history.replace(`/labbooks/${labbookName}`)
  }

  /**
  *  @param {event} e
  *  loads more labbooks using the relay pagination container
  */
  _loadMore = (e) => {
    isLoadingMore = true
    if(e){
      e.preventDefault();
    }
    this.props.relay.loadMore(
      10, // Fetch the next 10 feed items
      (ev) => {
        isLoadingMore = false;
      }
    );
  }

  render(){
      let {props} = this;

      if(props.feed.localLabbooks){

        return(
          <div className="LocalLabbooks">

            <WizardModal
              ref="wizardModal"
              handler={this.handler}
              history={this.props.history}
              {...props}
            />

            <div className="LocalLabbooks__title-bar flex flex--row justify--space-between">
              <h4 className="LocalLabbooks__title" onClick={()=> this.refs.wizardModal._showModal()} >
                Lab Books
                <div className="LocalLabbooks__title-add"></div>
              </h4>

            </div>
            <div className='LocalLabbooks__labbooks'>
              <div className="LocalLabbooks__sizer">
              <div
                key={'addLabbook'}
                onClick={()=> this.refs.wizardModal._showModal()}
                className="LocalLabbooks__panel LocalLabbooks__panel--add flex flex--row justify--center">
                <div
                  // onClick={()=> this._openImport()}
                  className="LocalLabbooks__labbook-icon">
                    <div className="LocalLabbooks__title-add"></div>
                </div>
              </div>

              <ImportModule
                  ref="ImportModule_localLabooks"
                  {...props}
                  className="LocalLabbooks__panel LocalLabbooks__panel--import" />

              {

                this.props.feed.localLabbooks.edges.map((edge) => {

                  return (
                    <LocalLabbookPanel
                      key={edge.node.name}
                      ref={'LocalLabbookPanel' + edge.node.name}
                      className="LocalLabbooks__panel"
                      edge={edge}
                      goToLabbook={this._goToLabbook}/>
                  )
                })
              }
            </div>

          </div>

        </div>
      )
    }else{
      return(<Loader />)
    }

  }
}

export default createPaginationContainer(
  LocalLabbooks,
  {feed: graphql`
      fragment LocalLabbooks_feed on Query{
        localLabbooks(first: $first, after:$cursor)@connection(key: "LocalLabbooks_localLabbooks"){
          edges {
            node {
              name
              description
              owner{
                id
                username
              }
              environment{
                id
                imageStatus
                containerStatus
              }
            }
            cursor
          }
          pageInfo {
            endCursor
            hasNextPage
            hasPreviousPage
            startCursor
          }
        }
      }
    `,
  },
  {
    direction: 'forward',
    getConnectionFromProps(props) {
      return props.feed.localLabbooks
    },
    getFragmentVariables(prevVars, first, cursor) {
      return {
        ...prevVars,
        first: first
      };
    },
    getVariables(props, {first, cursor}, fragmentVariables) {
      first = 10;
      cursor = props.feed.localLabbooks.pageInfo.endCursor;
      return {
        first,
        cursor
        // in most cases, for variables other than connection filters like
        // `first`, `after`, etc. you may want to use the previous values.
      };
    },
    query: graphql`
      query LocalLabbooksPaginationQuery(
        $first: Int!
        $cursor: String
      ) {
          ...LocalLabbooks_feed
      }
    `
  }
);
