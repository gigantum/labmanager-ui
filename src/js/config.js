import tips from './tips'

const CONFIG = {
  'api': process.env.NODE_ENV,
  'navigation_items': [
    {id:'overview', name: 'Overview'},
    {id:'activity', name: 'Activity', 'fragment': '...Activity_labbook'},
    {id:'environment', name: 'Environment', 'fragment': '...Environment_labbook'},
    {id:'code', name: 'Code'},
    {id:'inputData', name: 'Input Data'},
    {id:'outputData', name: 'Output Data'}
  ],
  'modalNav': [
    {'id': 'createLabook', 'description': 'Title & Description'},
    {'id': 'selectBaseImage', 'description': 'Base Image'},
    {'id': 'selectDevelopmentEnvironment', 'description': 'Dev Environment'},
    {'id': 'addEnvironmentPackage', 'description': 'Add Dependencies'},
    {'id': 'successMessage', 'description': 'Success'}
  ],
  'months': [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
  ],
  'fileBrowser':{
    'excludedFiles': [
      'DS_Store',
      'zip',
      'lbk',
      'pyc',
      'gitkeep'
    ]
  },

  containerStatus:{
    canEditEnvironment: (status) => {
      return (status === 'Stopped') || (status === 'Rebuild')
    }
  },
  userAPI: {
    getUsersQueryString: (userInput) => {
      const sanitizedUserInput = userInput.replace(/-/g, ' ')
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${sanitizedUserInput}*&q.options={fields: ['username^5','name']}&size=10`

      return encodeURI(apiURL)
    },
    getUserEmailQueryString: (email) => {
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${email}&q.options={fields: ['email']}&size=10`

      return encodeURI(apiURL)
    }
  },
  getToolTipText: (section) =>{
    return tips[section]
  },
  demoHostName: 'try.gigantum.com',
}

export default CONFIG
