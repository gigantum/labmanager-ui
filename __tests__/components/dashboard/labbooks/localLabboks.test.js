import React from 'react'
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import history from 'JS/history'
import {StaticRouter, Link} from 'react-router';
import config from './config'
import LocalLabbooks from 'Components/dashboard/labbooks/LocalLabbooks';
import relayTestingUtils from 'relay-testing-utils'
import {MemoryRouter } from 'react-router-dom'

import environment from 'JS/createRelayEnvironment'

const fixtures = {
  localLabbooks: config.data.localLabbooks
}

test('Test LocalLabbooks rendering', () => {

  const localLabbooks = renderer.create(

     relayTestingUtils.relayWrap(<LocalLabbooks history={history} {...fixtures} feed={config.data}/>)

  );

  const tree = localLabbooks.toJSON()

  expect(tree).toMatchSnapshot()
})


describe('Test LocalLabbooks panel length', () => {

  const localLabbooks = shallow(

     <LocalLabbooks history={history} {...fixtures} feed={config.data}/>

  );

  expect(localLabbooks.find('.LocalLabbooks__panel')).toHaveLength(21)
})


describe('Test LocalLabbooks load more', () => {
  const relay = {loadMore: () => {}}
  const localLabbooks = mount(

     <LocalLabbooks relay={relay} history={history} {...fixtures} feed={config.data}/>

  );

  //localLabbooks.find('button').at(0).simulate('click')
  //console.log(localLabbooks.find('.LocalLabbooks__panel'))
  expect(localLabbooks.find('.LocalLabbooks__panel')).toHaveLength(21)
})

describe('LocalLabbooks show modal', () => {
  const relay = {loadMore: () => {}}
  const localLabbooks = mount(

     <LocalLabbooks relay={relay} history={history} {...fixtures} feed={config.data}/>
  );

  // localLabbooks.refs = {wizardModal: {_showModal: () => {}}}
  localLabbooks.find('.LocalLabbooks__title').simulate('click')

  expect(localLabbooks.node.refs.wizardModal.state.modal_visible).toBeTruthy()

})

describe('LocalLabbooks show modal by panel', () => {
  const relay = {loadMore: () => {}}
  const localLabbooks = mount(

     <LocalLabbooks relay={relay} history={history} {...fixtures} feed={config.data}/>
  );
  // localLabbooks.refs = {wizardModal: {_showModal: () => {}}}
  localLabbooks.find('.LocalLabbooks__panel--add').simulate('click')

  expect(localLabbooks.node.refs.wizardModal.state.modal_visible).toBeTruthy()

})



describe('Test LocalLabbooks click', () => {

  const localLabbooks = mount(

      <LocalLabbooks history={history} feed={config.data}/>

  );

  localLabbooks.at(0).simulate('click')
  localLabbooks.setState({'labbookName': config.data.localLabbooks.edges[0].node.name})

  expect(localLabbooks.node.state.labbookName === config.data.localLabbooks.edges[0].node.name).toBeTruthy();
})


describe('Test LocalLabbooks edges output', () => {

  const localLabbooks = mount(

      <LocalLabbooks history={history} feed={config.data}/>

  );

  expect(localLabbooks.find('.LocalLabbooks__labbooks .LocalLabbooks__text-row h4').at(0).text()).toEqual(config.data.localLabbooks.edges[0].node.name)

})
