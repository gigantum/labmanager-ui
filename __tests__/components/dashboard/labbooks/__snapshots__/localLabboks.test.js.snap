// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LocalLabbooks rendering 1`] = `
<div
  className="LocalLabbooks"
>
  <div
    className="WizardModal"
  >
    <div
      className="WizardModal__modal hidden"
    >
      <div
        className="WizardModal__progress"
      >
        <ul
          className="WizardModal__progress-bar"
        >
          <li
            className="WizardModal__progress-item selected"
          >
            Title & Description
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
          <li
            className="WizardModal__progress-item"
          >
            
          </li>
        </ul>
      </div>
      <h4
        className="WizardModal__title"
      >
        Create a Lab Book
      </h4>
      <div
        className="WizardModal__modal-close"
        onClick={[Function]}
      >
        X
      </div>
      <div
        className="CreateLabbook"
      >
        <div
          className="CreateLabbook__modal-inner-container flex flex--column justify--space-between"
        >
          <div>
            <label>
              Title
            </label>
            <input
              className=""
              onChange={[Function]}
              placeholder="Enter a unique, descriptive title"
              type="text"
            />
            <span
              className="hidden"
            >
              Error: Title may only contain alphanumeric characters separated by hyphens. (e.g. lab-book-title)
            </span>
          </div>
          <div>
            <label>
              Description
            </label>
            <textarea
              className="CreateLabbook__description-input"
              onChange={[Function]}
              placeholder="Briefly describe this lab book, its purpose and any other key details. "
              type="text"
            />
          </div>
          <div
            className="CreateLabbook__text-divider-container"
          >
            <span
              className="CreateLabbook__text-divider"
            >
              or
            </span>
          </div>
          <div>
            <label>
              Add public Lab Books
            </label>
            <input
              placeholder="Enter URL Location"
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="flex flex--row justify--center"
      >
        <button
          className="WizardModal__progress-button flat--button"
          disabled={true}
          onClick={[Function]}
        >
          Previous
        </button>
        <button
          className="WizardModal__progress-button flat--button"
          onClick={[Function]}
        >
          Cancel
        </button>
        <button
          className="WizardModal__progress-button flat--button"
          disabled={true}
          onClick={[Function]}
        >
          skip
        </button>
        <button
          disabled={true}
          onClick={[Function]}
        >
          Save and Continue Setup
        </button>
      </div>
    </div>
  </div>
  <h4
    className="LocalLabbooks__title"
    onClick={[Function]}
  >
    Lab Books 
    <div
      className="LocalLabbooks__title-add"
    />
  </h4>
  <div
    className="LocalLabbooks__labbooks flex flex--row flex--wrap justify--left"
  >
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          RERE
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          sass
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          aaa-labbook
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which dont look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isnt anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adaddas324234
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asddsa
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adadsdas34343
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asdads
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adasads
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          sadads
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          addsasad
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asddasdsa
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adsasad
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          sadsad
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adsasdadas
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asddasad
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adsdasads3234
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          adsdas
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          adssaads
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          aadsdas
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asadsds
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          adsad
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdada
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asa
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdads
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asdas
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdadsdas
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdasdadsdas322332
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          223
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asddasadsasd21331
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asdads
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdds
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asdad
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdsadad
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          asdasd
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          asdsdasd
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          dasdas
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel flex flex--column justify--space-between"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__icon-row"
      >
        <div
          className="LocalLabbooks__labbook-icon"
        />
      </div>
      <div
        className="LocalLabbooks__text-row"
      >
        <h4>
          astronomicle
        </h4>
        <p
          className="LocalLabbooks__description"
        >
          
        </p>
      </div>
      <div
        className="LocalLabbooks__info-row flex flex--row"
      >
        <div
          className="LocalLabbooks__owner flex flex--row"
        >
          <div>
            Owner
          </div>
          <div
            className="LocalLabbooks__owner-icon"
          />
        </div>
        <div
          className="LocalLabbooks__status"
        />
      </div>
    </div>
    <div
      className="LocalLabbooks__panel LocalLabbooks__panel--add flex flex--row justify--center"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__labbook-icon"
      >
        <div
          className="LocalLabbooks__title-add"
        />
      </div>
    </div>
  </div>
</div>
`;
